File: src/java_package/Hyperbola.java
package java_package;

public class Hyperbola extends Curve {
    private double a; // hyperbola's a coefficient
    private double b; // hyperbola's b coefficient

    public Hyperbola(double a, double b) {
        this.a = a;
        this.b = b;
    }

    @Override
    double calculateY(double x) {
        return Math.sqrt((Math.pow(x, 2) / Math.pow(a, 2)) - 1) * b;
    }
}

File: src/kotlin_package/Hyperbola.kt
package kotlin_package

import kotlin.math.pow
import kotlin.math.sqrt

class Hyperbola(private var a: Double, private var b: Double) : Curve() {
    @Override
    override fun calculateY(x: Double): Double {
        return sqrt((x.pow(2) / a.pow(2)) - 1) * b
    }
}